examples/server-communication__offline/cypress/integration/offline-spec.js

```diff
@@ -1,177 +1,87 @@
+    })
   })
 
-  it('makes fetch request when offline', () => {
-    cy.visit('/')
-
-    goOffline()
-    assertOffline()
-
-    // let's spy on the "fetch" method the app calls
-    cy.window().then((w) => cy.spy(w, 'fetch').withArgs(`${url}?_limit=3`).as('fetchUsers'))
-
-    cy.get('#load-users').click()
-    cy.get('@fetchUsers').should('have.been.calledOnce')
-
-    // now let's go back online and fetch the users
-    goOnline()
-    assertOnline()
-    cy.get('#load-users').click()
-    cy.get('.user').should('have.length', 3)
-    cy.get('@fetchUsers').should('have.been.calledTwice')
-  })
+  context('when offline', () => {
+    beforeEach(() => {
+      cy.visit('/', {
+        onBeforeLoad(win) {
+          cy.stub(win.navigator, 'onLine', false).as('online')
+        }
+      })
+      cy.intercept('*', { forceNetworkError: true }).as('users')
+    })
 
-  it('does not reach the outside network when offline', () => {
-    cy.visit('/')
+    it('shows network status as offline', () => {
+      cy.contains('#network-status', 'offline')
+    })
 
-    // before we go offline we have to set up network intercepts
-    // since they need to be communicated outside the browser
-    // and lets keep track the number of network calls made
-    let callCount = 0
+    it('shows error if we stub the network call', () => {
+      cy.get('#load-users').click()
+      cy.contains('#users', 'Problem fetching users')
 
-    cy.intercept(`${url}*`, () => {
-      callCount += 1
-    }).as('users')
+      cy.wait('@users') // the network call happens
+    })
 
-    goOffline()
-    assertOffline()
+    it('shows error trying to fetch users in offline mode', () => {
+      cy.get('#load-users').click()
+      cy.contains('#users', 'Problem fetching users')
+    })
 
-    cy.get('#load-users').click()
-    cy.contains('#users', 'Problem fetching users Failed to fetch')
+    it('makes fetch request when offline', () => {
+      // let's spy on the "fetch" method the app calls
+      cy.window().then((w) => cy.spy(w, 'fetch').withArgs(`${url}?_limit=3`).as('fetchUsers'))
 
-    // the cy.intercept network call does NOT happen
-    // because the browser does not fire it
-    // and thus our network proxy does not see it
-    cy.then(() => {
-      expect(callCount, 'no network calls made').to.equal(0)
+      cy.get('#load-users').click()
+      cy.get('@fetchUsers').should('have.been.calledOnce')
     })
 
-    // now let's go back online and fetch the users
-    goOnline()
-    assertOnline()
+    it('does not reach the outside network when offline', () => {
+      let callCount = 0
 
-    cy.get('#load-users').click()
-    // we can retry the assertion to know when the network call has happened
-    // using .should callback function with an assertion inside
-    .should(() => {
-      expect(callCount, 'single network call').to.equal(1)
-    })
+      cy.intercept('*', { forceNetworkError: true }, () => {
+        callCount += 1
+      }).as('users')
```

Comment:
this example is incorrect. in this example, the third argument is never recognized as anything. you've invoked this overload of cy.intercept:

```js
cy.intercept(url, routeHandler)
```

no third argument...

really, we should throw an error when it's used like this

Reply:
@jennifer-shehane i'm not sure why this was marked as resolved, this example is wrong

Reply:
nvm, somehow i missed ur commit :facepalm: